### ISSUE TITLE
Steam crashing on launch

### ISSUE THREAD

**Issue by ohno1052** [flathub/com.valvesoftware.Steam#1273]
**Game information**
Issue with steam itself

**Distribution name and version where applicable**
Fedora Silverblue 39

**Flatpak info**
Flatpak 1.15.6
nvidia-550-54-14
default
host

**Problem description**
Tried to create, rename, change, permissions, delete - nothing changed.
Reinstallation with removal of all data doesn't help.
[log.txt](https://github.com/flathub/com.valvesoftware.Steam/files/14600680/log.txt)

**Does this issue reproduce with native Steam**
No

**Comment by Constancies** [comment]
I'm having a similar issue.

Logs:
```
flatpak run com.valvesoftware.Steam
INFO:root:https://github.com/flathub/com.valvesoftware.Steam/wiki
ERROR:root:~/.steam/root isn't a symlink to an existing directory, cannot determine current prefix
INFO:root:Will set XDG dirs prefix to /home/constancies/.var/app/com.valvesoftware.Steam
Traceback (most recent call last):
  File "/app/bin/steam", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/app/lib/python3.11/site-packages/steam_wrapper.py", line 486, in main
    should_restart = migrate_config(current_info, xdg_dirs_prefix)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/steam_wrapper.py", line 316, in migrate_config
    ignore = _get_host_xdg_mounts("xdg-config", flatpak_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/steam_wrapper.py", line 305, in _get_host_xdg_mounts
    dirs.add(os.path.join(*path_seq[1:]))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: join() missing 1 required positional argument: 'a'

```

**Comment by jntesteves** [comment]
@Constancies I think you found a bug, yes. From what I understand from the code where you hit this error, it's only possible to reach this part if you have the permission `--filesystem=xdg-config`, sharing the entire `xdg-config` without specifying a subdirectory like `xdg-config/MangoHud:ro`, etc. That is giving up security excessively and currently not supported (at least not until this bug is fixed). This is a different bug from OP, please file a new issue for it.

About OP, I took a quick look, but I'm not sure yet if it's a bug in the wrapper script or a problem in your setup why the resource is busy. Needs further investigation.
